ifupdown2/addons/auto.py

```diff
@@ -0,0 +1,161 @@
+
+    import ifupdown2.ifupdown.policymanager as policymanager
+    import ifupdown2.ifupdown.ifupdownflags as ifupdownflags
+
+    from ifupdown2.ifupdown.iface import *
+    from ifupdown2.ifupdown.utils import utils
+
+    from ifupdown2.ifupdownaddons.modulebase import moduleBase
+except (ImportError, ModuleNotFoundError):
+    from lib.addon import Addon
+    from lib.log import LogManager
+
+    import ifupdown.policymanager as policymanager
+    import ifupdown.ifupdownflags as ifupdownflags
+
+    from ifupdown.iface import *
+    from ifupdown.utils import utils
+
+    from ifupdownaddons.modulebase import moduleBase
+
+
+class auto(Addon, moduleBase):
+    """ ifupdown2 addon module to configure slaac on inet6 interface """
+
+    def __init__(self, *args, **kargs):
+        Addon.__init__(self)
+        moduleBase.__init__(self, *args, **kargs)
+
+    def syntax_check(self, ifaceobj, ifaceobj_getfunc):
+        return self.is_auto_allowed_on(ifaceobj, syntax_check=True)
+
+    def is_auto_allowed_on(self, ifaceobj, syntax_check):
+        if ifaceobj.addr_method and 'auto' in ifaceobj.addr_method:
+            return utils.is_addr_ip_allowed_on(ifaceobj, syntax_check=True)
+        return True
+
+    def _up(self, ifaceobj):
+
+        if ifaceobj.link_privflags & ifaceLinkPrivFlags.KEEP_LINK_DOWN:
+            self.logger.info("%s: skipping auto configuration: link-down yes" % ifaceobj.name)
+            return
+
+        try:
+            if 'inet6' in ifaceobj.addr_family:
+                inet6conf = self.cache.get_link_inet6_conf(ifaceobj.name)
+                if 'accept_ra' in inet6conf and inet6conf['accept_ra'] != 1:
+                    self.sysctl_set('net.ipv6.conf.%s' %ifaceobj.name +
+                            '.accept_ra', 1)
+                if 'autoconf' in inet6conf and inet6conf['autoconf'] != 1:
+                    self.sysctl_set('net.ipv6.conf.%s' %ifaceobj.name +
+                           '.autoconf', 1)
+
+        except Exception as e:
+            self.logger.error("%s: %s" % (ifaceobj.name, str(e)))
+            ifaceobj.set_status(ifaceStatus.ERROR)
+
+    def _down(self, ifaceobj):
+        if 'inet6' in ifaceobj.addr_family:
+            self.cache.force_address_flush_family(ifaceobj.name, socket.AF_INET6)
+        self.netlink.link_down(ifaceobj.name)
+
+    def _query_check(self, ifaceobj, ifaceobjcurr):
+        if not self.cache.link_exists(ifaceobj.name):
+            return
+        ifaceobjcurr.addr_family = ifaceobj.addr_family
+        ifaceobjcurr.addr_method = 'auto'
+
+        inet6conf = self.cache.get_link_inet6_conf(ifaceobj.name)
+        if inet6conf['accept_ra'] == 1 and inet6conf['autoconf'] == 1:
+            ifaceobjcurr.status = ifaceStatus.SUCCESS
+        else:
+            ifaceobjcurr.status = ifaceStatus.ERROR
+
+    def _query_running(self, ifaceobjrunning):
+        if not self.cache.link_exists(ifaceobjrunning.name):
+            return
```

Comment:
No need to define query_running if the method stays empty

Reply:
mmm, seem to be needed, as module can't load without it

warning: cannot load "auto" module: name '_query_running' is not defined

Reply:
Oh i see, in that case lets just keep the function empty like that:

```
def _query_running(self, ifaceobjrunning):
    pass
```

and someday i'll fix the `'_query_running' is not defined` error

Reply:
fixed